editor/plugins/particle_process_material_editor_plugin.cpp

```diff
@@ -0,0 +1,471 @@
+			max_edit->set_read_only(max_edit->get_max() == 0);
+			max_edit->set_block_signals(false);
+		} break;
+	}
+}
+
+void ParticleProcessMaterialMinMaxPropertyEditor::_sync_sliders(float, const EditorSpinSlider *p_changed_slider) {
+	switch (slider_mode) {
+		case Mode::RANGE: {
+			if (p_changed_slider == max_edit) {
+				min_edit->set_value_no_signal(MIN(min_edit->get_value(), max_edit->get_value()));
+			}
+			min_range->set_value(min_edit->get_value());
+			if (p_changed_slider == min_edit) {
+				max_edit->set_value_no_signal(MAX(min_edit->get_value(), max_edit->get_value()));
+			}
+			max_range->set_value(max_edit->get_value());
+			_sync_property();
+		} break;
+
+		case Mode::MIDPOINT: {
+			if (p_changed_slider == min_edit) {
+				max_edit->set_block_signals(true); // If max changes, value may change.
+				max_edit->set_max(_get_max_spread());
+				max_edit->set_read_only(max_edit->get_max() == 0);
+				max_edit->set_block_signals(false);
+			}
+			min_range->set_value(min_edit->get_value() - max_edit->get_value());
+			max_range->set_value(min_edit->get_value() + max_edit->get_value());
+			_sync_property();
+		} break;
+	}
+
+	property_range.x = MIN(min_range->get_value(), min_range->get_min());
+	property_range.y = MAX(max_range->get_value(), max_range->get_max());
+}
+
+float ParticleProcessMaterialMinMaxPropertyEditor::_get_max_spread() const {
+	float max_spread = max_range->get_max() - min_range->get_min();
+
+	if (max_edit->is_greater_allowed()) {
+		return max_spread;
+	}
+
+	if (!min_edit->is_lesser_allowed()) {
+		max_spread = MIN(max_spread, min_edit->get_value() - min_edit->get_min());
+	}
+
+	if (!min_edit->is_greater_allowed()) {
+		max_spread = MIN(max_spread, min_edit->get_max() - min_edit->get_value());
+	}
+
+	return max_spread;
+}
+
+void ParticleProcessMaterialMinMaxPropertyEditor::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_THEME_CHANGED: {
+			toggle_mode_button->set_icon(get_editor_theme_icon(SNAME("Anchor")));
+			range_slider_left_icon = get_editor_theme_icon(SNAME("RangeSliderLeft"));
+			range_slider_right_icon = get_editor_theme_icon(SNAME("RangeSliderRight"));
+
+			min_edit->add_theme_color_override(SNAME("label_color"), get_theme_color(SNAME("property_color_x"), EditorStringName(Editor)));
+			max_edit->add_theme_color_override(SNAME("label_color"), get_theme_color(SNAME("property_color_y"), EditorStringName(Editor)));
+
+			const bool dark_theme = EditorSettings::get_singleton()->is_dark_theme();
+			const Color accent_color = get_theme_color(SNAME("accent_color"), EditorStringName(Editor));
+			background_color = dark_theme ? Color(0.3, 0.3, 0.3) : Color(0.7, 0.7, 0.7);
+			normal_color = dark_theme ? Color(0.5, 0.5, 0.5) : Color(0.8, 0.8, 0.8);
+			hovered_color = dark_theme ? Color(0.8, 0.8, 0.8) : Color(0.6, 0.6, 0.6);
+			drag_color = hovered_color.lerp(accent_color, 0.8);
+			midpoint_color = dark_theme ? Color(1, 1, 1) : Color(0, 0, 0);
```

Comment:
We need to put stuff like this into the editor theme. But we don't have a good reference how to do this in an organized manner. A task for another day...